src/librustc/middle/cstore.rs

```diff
@@ -42,6 +42,28 @@ pub use self::NativeLibraryKind::*;
 
 pub const METADATA_FILENAME: &str = "rust.metadata.bin";
 
+pub fn metadata_section_name(target: &Target) -> &'static str {
+    // Historical note:
+    //
+    // When using link.exe it was seen that the section name `.note.rustc`
+    // was getting shortened to `.note.ru`, and according to the PE and COFF
+    // specification:
+    //
+    // > Executable images do not use a string table and do not support
+    // > section names longer than 8 characters
+    //
+    // https://msdn.microsoft.com/en-us/library/windows/hardware/gg463119.aspx
+    //
+    // As a result, we choose a slightly shorter name! As to why
+    // `.note.rustc` works on MinGW, that's another good question...
+
+    if target.options.is_like_osx {
+        "__DATA,.rustc"
+    } else {
+        ".rustc"
```

Comment:
Can we revise the commit message? I thought that you were putting the metadata in a section called `.cstore`, but I now realise you meant you were transplanting code into `cstore.rs`.

This code is borrowed from elsewhere, right?

Reply:
I'll change the commit message when I rebase.